invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/UserDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -0,0 +1,176 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { generatePath } from 'react-router';
+import {
+  Container,
+  Grid,
+  Segment,
+  Icon,
+  Header,
+  Button,
+  Input,
+} from 'semantic-ui-react';
+import { BackOfficeURLS } from '../../../../../common/urls';
+import { Error, Loader } from '../../../../../common/components';
+import { ResultsList as ItemsResultsList } from './components';
+import _isEmpty from 'lodash/isEmpty';
+
+export default class ItemsCheckout extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchItems = this.props.fetchItems;
+    this.updateQueryString = this.props.updateQueryString;
+    this.clearResults = this.props.clearResults;
+    this.checkoutItem = this.props.checkoutItem;
+    this.fetchPatronCurrentLoans = this.props.fetchPatronCurrentLoans;
+  }
+
+  componentDidMount() {
+    if (this.searchInput) {
+      this.searchInput.focus();
+    }
+  }
+
+  onInputChange = queryString => {
+    this.updateQueryString(queryString);
+  };
+
+  executeSearch = queryString => {
+    if (queryString) {
+      return this.fetchItems(queryString);
+    } else {
+      return this.fetchItems(this.props.queryString);
+    }
+  };
+
+  _renderSearchBar = () => {
+    return (
+      <Input
+        action={{
+          content: 'Search',
+          onClick: () => {
+            this.executeSearch();
+          },
+        }}
+        fluid
+        focus
+        placeholder={'Search by barcode...'}
+        onChange={(e, { value }) => {
+          this.onInputChange(value);
+        }}
+        onPaste={e => {
+          let queryString = e.clipboardData.getData('Text');
```

Comment:
I am a bit concerned this implementation, it has to be re-written. In first place `checkoutItem` doesn't have to wait for `executeSearch` since it doesn't have any dependencies on it. The arbitrary `setTimeout` thing gave me slight goosebumps on my left hand. It looks like a complicated task you are trying to achieve there, we can walk through it together and make it better :)

Reply:
it works like this because when you refresh the current loans table immediately it is not indexed yet. 
also checkout item has dependency on it. What if there is a problem with the barcode and the item does not exist? (Although I forgot to handle here this case)